typehints/common/digi/ble.pyi

```diff
@@ -393,3 +480,589 @@ def gap_scan(
+    XBee will continue to be used in servicing the connection until the
+    connection object is deleted, either by soft-resetting the MicroPython REPL,
+    or by garbage collection. (Note: Manually triggering garbage collection by
+    using ``gc.collect()`` is typically not necessary in MicroPython.)
+
+    Instead of needing to call ``close()`` directly, you may instead use the
+    object returned by ``gap_connect`` as a context manager. By doing this,
+    when the ``with`` block is exited, the GAP connection is automatically closed.
+    This approach uses less code and is less error-prone, because ``close()`` will
+    be called even if an exception is raised.
+
+    For example, if you want to connect to a peripheral, discover and print its
+    services, then close the connection, you can do this as follows::
+
+        def show_services(addr_type, address):
+            with ble.gap_connect(addr_type, address) as peripheral:
+                for service in peripheral.gattc_services():
+                    print(service)
+
+        show_services(ble.ADDR_TYPE_PUBLIC, mac)
+
+    Compare to this example which does not use a context manager. Note the use of
+    a try/finally block, and the need to call ``close()`` explicitly::
+
+        def show_services(addr_type, address):
+            peripheral = ble.gap_connect(addr_type, address)
+            try:
+                for service in peripheral.gattc_services():
+                    print(service)
+            finally:
+                # Make sure to call close() even if an exception is raised.
+                peripheral.close()
+
+        show_services(ble.ADDR_TYPE_PUBLIC, mac)
+
+    **gap_scan and gap_connect:**
+
+    Any ongoing GAP scan operation (see ``gap_scan()``) will block a GAP
+    connection request until the scan completes. If the scan does not complete
+    within ``timeout_ms``, the ``gap_connect`` call will raise OSError ETIMEDOUT.
+
+    Starting a GAP scan operation while a GAP connection is open is allowed
+    and does not affect the existing GAP connection.
+
+    **timeout_ms parameter:**
+
+    The ``timeout_ms`` parameter specifies the maximum length of time that the
+    ``gap_connect`` call should be allowed to block waiting for the connection
+    to be established. Note that per the Bluetooth Core Specification, if the
+    remote device does not respond to a connection request within six (6)
+    connection intervals, the connection shall be considered lost.
+    In other words, the connection attempt can time out in less time than
+    specified in ``timeout_ms``, depending on the connection interval used.
+    See the ``interval_ms`` parameter for ``digi.ble.config()`` for more
+    information on the connection interval setting.
+
+    **Note:**
+
+    The ``gap_connect`` function is new in the following firmware versions:
+        * XBee3 Zigbee: version 100A
+        * XBee3 802.15.4: version 2004
```

Comment:
This is a typo (2004 did not have `gap_connect`). I'll need to look these over and make sure we're using the correct versions here.